modules/server/src/topics/errorlogtopic.cpp

```diff
@@ -0,0 +1,129 @@
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/server/include/topics/errorlogtopic.h>
+
+#include <ghoul/logging/callbacklog.h>
+#include <ghoul/logging/logmanager.h>
+
+namespace {
+    constexpr std::string_view StartSubscription = "start_subscription";
+    constexpr std::string_view StopSubscription = "stop_subscription";
+
+    constexpr std::string_view SettingsKey = "settings";
+    constexpr std::string_view TimeStampingKey = "timeStamping";
+    constexpr std::string_view DateStampingKey = "dateStamping";
+    constexpr std::string_view CategoryStampingKey = "categoryStamping";
+    constexpr std::string_view LogLevelStampingKey = "logLevelStamping";
+    constexpr std::string_view LogLevelKey = "logLevel";
+}
+
+namespace openspace {
+
+ErrorLogTopic::~ErrorLogTopic()
+{
+    if (_log) {
+        ghoul::logging::LogManager::ref().removeLog(_log);
+        _log = nullptr;
+    }
+}
+
+void ErrorLogTopic::handleJson(const nlohmann::json& json)
+{
+    const std::string& event = json.at("event").get<std::string>();
+
+    if (event == StartSubscription) {
+        // Default settings for logging
+        auto timeStamping = ghoul::logging::Log::TimeStamping::Yes;
+        auto dateStamping = ghoul::logging::Log::DateStamping::Yes;
+        auto categoryStamping = ghoul::logging::Log::CategoryStamping::Yes;
+        auto logLevelStamping = ghoul::logging::Log::LogLevelStamping::Yes;
+        auto logLevel = ghoul::logging::LogLevel::AllLogging;
+
+        // Check if we got log settings on subscription
+        auto settings = json.find(SettingsKey);
+
+        if (settings != json.end()) {
+            auto ts = settings->find(TimeStampingKey);
+            if (ts != settings->end()) {
```

Comment:
Suggestion (if it fits):
```suggestion
            if (auto ts = settings->find(TimeStampingKey);  ts != settings->end()) {
```